src/main/perl/Path.pm

```diff
@@ -544,6 +565,103 @@ sub directory
 }
 
 
+=item _make_link
+
+This method is mainly a wrapper over C<LC::Check::link>
+returning the standard CAF::Path return values. Every option
+supported by C<LC::Check::link> is supported. C<NoActionFlag>
+is handled by C<LC::Check::link>.
+
+This in internal method, not supposed to be called directly.
+Either call C<symlink> or C<hardlink> public methods instead.
+
+=cut
+
+sub _make_link
+{
+    my ($self, $target, $link_path, %opts) = @_;
+    my $link_type = $opts{hard} ? "hardlink" : "symlink";
+
+    $link_path = $self->_untaint_path($link_path, $link_type) || return;
+    $target = $self->_untaint_path($target, $link_type) || return;
+
+    $self->_reset_exception_fail($link_type);
+
+    my $status = $self->LC_Check('link', [$link_path, $target], \%opts);
+            
+    if ( defined($status) ) {
+        return ($status ? CHANGED : SUCCESS);
+    } else {
+        return;
+    }
+}
+
+=item hardlink
+
+Create a hardlink C<link_path> whose target is C<target>.
+
+On failure, returns undef and sets the fail attribute.
+If C<link_path> exists and is a file, it is updated. 
+C<target> must exist and must reside in the same file system
```

Comment:
Yes you can! Only directories are not supported because not all file systems support hardlink to directories (see `man ln`, in fact `ln` need special options (`-d -F`) restricted to `root`). Example showing that it works for symlinks:
```
lx3/jouvin % echo "" > a_file
lx3/jouvin % ls -i a_file 
68 a_file
lx3/jouvin % ln -s a_file a_symlink
lx3/jouvin % ls -i a_symlink 
149 a_symlink
lx3/jouvin % ln a_symlink a_hardlink
lx3/jouvin % ls -i a_hardlink 
149 a_hardlink
```
You can see that `a_hardlink` has the same inode number as `a_symlink`(and is different from `a_file`).